src/librustc/mir/repr.rs

```diff
@@ -144,6 +147,13 @@ impl<'tcx> Mir<'tcx> {
     pub fn predecessors_for(&self, bb: BasicBlock) -> Ref<Vec<BasicBlock>> {
         Ref::map(self.predecessors(), |p| &p[bb])
     }
+
+    #[inline]
```

Comment:
An invalidate-on-any-change cache could be nice for preventing duplicate uses between passes.